hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ha/OMHAMetrics.java

```diff
@@ -0,0 +1,132 @@
+import org.apache.hadoop.metrics2.MetricsCollector;
+import org.apache.hadoop.metrics2.MetricsInfo;
+import org.apache.hadoop.metrics2.MetricsSource;
+import org.apache.hadoop.metrics2.MetricsRecordBuilder;
+import org.apache.hadoop.metrics2.annotation.Metrics;
+import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
+import org.apache.hadoop.metrics2.lib.Interns;
+import org.apache.hadoop.metrics2.lib.MetricsRegistry;
+import org.apache.hadoop.ozone.OzoneConsts;
+
+/**
+ * Class to maintain metrics and info related to OM HA.
+ */
+@Metrics(about = "OzoneManager HA Metrics", context = OzoneConsts.OZONE)
+public final class OMHAMetrics implements MetricsSource {
+
+  /**
+   * Private nested class to hold the values
+   * of MetricsInfo for OMHAMetrics.
+   */
+  private static final class OMHAMetricsInfo {
+
+    private static final MetricsInfo OZONE_MANAGER_HA_LEADER_STATE =
+        Interns.info("OzoneManagerHALeaderState",
+            "Leader active state of OzoneManager node (1 leader, 0 follower)");
+
+    private static final MetricsInfo NODE_ID =
+        Interns.info("NodeId", "OM node Id");
+
+    private long ozoneManagerHALeaderState;
+    private String nodeId;
+
+    OMHAMetricsInfo() {
+      this.ozoneManagerHALeaderState = 0L;
+      this.nodeId = "";
+    }
+
+    public long getOzoneManagerHALeaderState() {
+      return ozoneManagerHALeaderState;
+    }
+
+    public void setOzoneManagerHALeaderState(long ozoneManagerHALeaderState) {
+      this.ozoneManagerHALeaderState = ozoneManagerHALeaderState;
+    }
+
+    public String getNodeId() {
+      return nodeId;
+    }
+
+    public void setNodeId(String nodeId) {
+      this.nodeId = nodeId;
+    }
+  }
+
+  public static final String SOURCE_NAME =
+      OMHAMetrics.class.getSimpleName();
+  private final OMHAMetricsInfo omhaMetricsInfo = new OMHAMetricsInfo();
+  private MetricsRegistry metricsRegistry;
+
+  private String currNodeId;
+  private String leaderId;
+
+  private OMHAMetrics(String currNodeId, String leaderId) {
+    this.currNodeId = currNodeId;
+    this.leaderId = leaderId;
+    this.metricsRegistry = new MetricsRegistry(SOURCE_NAME);
+  }
+
+  /**
+   * Create and return OMHAMetrics instance.
+   * @return OMHAMetrics
+   */
+  public static OMHAMetrics create(
+      String nodeId, String leaderId) {
+    OMHAMetrics metrics = new OMHAMetrics(nodeId, leaderId);
+    return DefaultMetricsSystem.instance()
+        .register(SOURCE_NAME, "Metrics for OM HA", metrics);
+  }
+
+  /**
+   * Unregister the metrics instance.
+   */
+  public static void unRegister() {
+    DefaultMetricsSystem.instance().unregisterSource(SOURCE_NAME);
+  }
+
+  @Override
+  public synchronized void getMetrics(MetricsCollector collector, boolean all) {
+
+    MetricsRecordBuilder recordBuilder = collector.addRecord(SOURCE_NAME);
+
+    // Check current node state (1 leader, 0 follower)
+    int state = currNodeId.equals(leaderId) ? 1 : 0;
```

Comment:
If `leaderId` is an empty string due to leader unavail at metrics registration, we should probably check here and omit setting metrics.  Just return.